stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -3201,33 +3197,23 @@ final internal class _Native${Self}StorageKeyNSEnumerator<
 }
 #endif
 
-/// This class is an artifact of the COW implementation.  This class only
-/// exists to keep separate retain counts separate for:
-/// - `${Self}` and `NS${Self}`,
-/// - `${Self}Index`.
-///
-/// This is important because the uniqueness check for COW only cares about
-/// retain counts of the first kind.
-///
-/// Specifically, `${Self}` points to instances of this class.  This class
-/// is also a proper `NS${Self}` subclass, which is returned to Objective-C
-/// during bridging.  `${Self}Index` points directly to
-/// `_Native${Self}Storage`.
-final internal class _Native${Self}StorageOwner<${TypeParametersDecl}>
+#if _runtime(_ObjC)
+/// This class exists for Objective-C bridging. It holds a reference to a
+/// `_Native${Self}Storage`, and can be upcast to `NS${Self}` when
+/// bridging is necessary.
+final internal class _Native${Self}BridgingStorage<${TypeParametersDecl}>
```

Comment:
Could you provide more information about why this separate class is needed?  I have a suspicion that allocating a new instance of it every time might be a cause of the performance regression.

There might be a correctness issue as well, some Objective-C code does not work well if the same Swift Dictionary is bridged to a fresh pointer every time, because there is code out there that relies on pointer identity.  It is broken, but we have binary compatibility constraints, and everything just works better if we can guarantee a stable pointer.

Did you consider making the StorageImpl type conform to the `_SwiftNativeNS${Self}` protocols?